internal/service/accrual_service.go

```diff
@@ -0,0 +1,151 @@
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/golangTroshin/gophermat/internal/config"
+	"github.com/golangTroshin/gophermat/internal/model"
+	"gorm.io/gorm"
+)
+
+type APIResponse struct {
+	Order   string  `json:"order"`
+	Status  string  `json:"status"`
+	Accrual float64 `json:"accrual,omitempty"`
+}
+
+var (
+	pause    bool
+	pauseMux sync.Mutex
+)
+
+func fetchOrderStatus(orderNumber string) (*APIResponse, error) {
+	url := fmt.Sprintf("%s/api/orders/%s", config.Options.AccrualSystemAddress, orderNumber)
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusNoContent {
+		return nil, fmt.Errorf("order not found in the accrual system")
+	} else if resp.StatusCode == http.StatusTooManyRequests {
+		retryAfter := resp.Header.Get("Retry-After")
+		if retryAfter != "" {
+			retryAfterSeconds, err := strconv.Atoi(retryAfter)
+			if err != nil {
+				return nil, fmt.Errorf("error parsing Retry-After header: %v", err)
+			}
+
+			pauseMux.Lock()
+			pause = true
+			pauseMux.Unlock()
+
+			log.Printf("Too many requests, pausing for %d seconds", retryAfterSeconds)
+			time.Sleep(time.Duration(retryAfterSeconds) * time.Second)
+
+			pauseMux.Lock()
+			pause = false
+			pauseMux.Unlock()
```

Comment:
Мне кажется, можно попробовать немного оптимизировать, если использовать RWLock.

```suggestion
			{
				pauseMux.Lock()
				defer pauseMux.Unlock()
				
				pause = true
			
				log.Printf("Too many requests, pausing for %d seconds", retryAfterSeconds)
				time.Sleep(time.Duration(retryAfterSeconds) * time.Second)

				pause = false
			}
```